### PR TITLE
Add `.yaml` extension to generated file in OpenAPI build plugin

### PR DESCRIPTION
**Pull request by lnash94**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-build-extension/src/main/java/io/ballerina/openapi/build/HttpServiceAnalysisTask.java [hunk 1]
```diff
@@ -128,7 +130,8 @@ private void writeOpenAPIYaml(Path outPath, OASResult oasResult, List<Diagnostic
             try {
                 // Create openapi directory if not exists in the path. If exists do not throw an error
                 Files.createDirectories(Paths.get(outPath + OAS_PATH_SEPARATOR + OPENAPI));
-                String fileName = resolveContractFileName(outPath.resolve(OPENAPI), oasResult.getServiceName(),
+                String fileName = resolveContractFileName(outPath.resolve(OPENAPI),
+                        oasResult.getServiceName(),
                         false);
```
Comment (NipunaRanasinghe) [inline openapi-build-extension/src/main/java/io/ballerina/openapi/build/HttpServiceAnalysisTask.java:135]: Shall we fix formatting in here?

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/extension/build/BuildExtensionTests.java [hunk 2]
```diff
@@ -47,18 +49,17 @@ public void setupDistributions() throws IOException {
     }
 
     @Test(description = "Check openapi build plugin in `bal build` command")
-    public void annotationWithOutBuildOption() throws IOException {
+    public void onlyBuildCommand() throws IOException {
         executeCommand("project_1");
     }
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/extension/build/BuildExtensionTests.java:54]: Shouldn't we have assertions in here to ensure that any open API contracts are not generated in the `target` directory?

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/extension/build/BuildExtensionTests.java [hunk 3]
```diff
@@ -47,18 +49,17 @@ public void setupDistributions() throws IOException {
...
 
     @Test(description = "Check openapi build plugin in `bal build` command")
-    public void annotationWithOutBuildOption() throws IOException {
+    public void onlyBuildCommand() throws IOException {
```
Comment (NipunaRanasinghe) [inline openapi-integration-tests/src/test/java/io/ballerina/openapi/extension/build/BuildExtensionTests.java:52]: Shall we use more appropriate naming convention for the test cases? (This seems to be a negative test case to me). Better to refactor the also the other test case names in this class.

File: openapi-integration-tests/src/test/java/io/ballerina/openapi/extension/build/BuildExtensionTests.java (hunk 2 above)
Comment (lnash94) [reply openapi-integration-tests/src/test/java/io/ballerina/openapi/extension/build/BuildExtensionTests.java:54]: yeah we need to test, this assertion implementation working on separate PR. :)

**Review by NipunaRanasinghe — APPROVED** [review]
